eclipse_workflow.md

```diff
@@ -0,0 +1,305 @@
+
+
+
+Finally, we need to tell Eclipse about the build path.
+
+1. From the project context menu, choose `Build configurations > Manage ...`.
+Rename the `Build GNU` build configuration to according to the build directory (helps
+keeping an overview later), in our case `bld_master_nompi`.
+1. Choose the project in the project browser, then `Properties` from the context
+menu.
+1. Go to `C/C++ Build`
+1. It should show the`bld_master_nompi` (or whatever name you chose)  as active
+configuration. 
+1. Then, in the `Build location` section of the `C/C++ Build` window, click
+`File system ...`, then choose `$NEST_ROOT/bld_master_nompi`.
+1. If you want to build in parallel, remove the check for
+`Use default build command` and enter `make -j4` as build command
+  (replace 4 with a suitable number for your computer).
+
+Finally, we need to amend the search path for tools Eclipse uses. In the
+project properties browser,
+
+1. go to `C/C++ Build > Enviroment`
+1. click `Select ...` and choose `PATH`
+1. select `PATH` in the variables list and click `Edit ...`
+1. prepend to the path
+    1. `/usr/local/bin:` if you use Homebrew
+    1. `/opt/local/bin:` if you use MacPorts
+
+You can now build the project by choosing `Build project` from the
+context menu.
+
+To install or run the testsuite, you should add additional make
+targets:
+
+1. Go to the Context Menu of the project
+1. Choose `Make Targets > Create ...` and add a target,
+e.g. `install` by entering this as the target name.
+1. Remove the check for `Run all project builders`.
+1. You should create targets 
+    - `all` (builds nest)
+    - `install` (installs nest, including tests and help)
+	- `install-exec` (installs compiled code and Python, but not SLI
+	code, tests, or help; faster if you only changed C++ or Python files)
+    - `installcheck` (runs the testsuite)
+1. You can run the targets by choosing `Make Targets > Build ...` from
+the Context Menu.
+
+See also [https://wiki.eclipse.org/CDT/Autotools/User_Guide](https://wiki.eclipse.org/CDT/Autotools/User_Guide).
+
+### Running NEST from Eclipse
+
+To run NEST within Eclipse,
+
+1. go to the project properties browser
+1. select `Run/Debug Settings`
+1. select `NEST Build (GNU)` and click `Edit ...`
+1. rename to `run_master_nompi`
+1. under `C/C++ Application` click `Browse ...` and select
+`$NEST_ROOT/bld_master_nompi/ins/bin/nest`
+1. select `Disable auto build` (because that only builds, but does not install)
+
+You can now run NEST by clicking the "Play" button. Input is echoed in
+a slightly funny way in the build-in console, but NEST works fine. You
+need to quit NEST with the `quit` command, `Ctrl-D` does not seem to
+work (made my machine hang totally on one occasion).
+
+#### Running PyNEST from Eclipse
+
+To be written.
```

Comment:
very interested in seeing instructions for this. I'd love to be able to set up and start a simulation from python, and then debug it graphically (expanding objects in a tree view, viewing call stack, etc.) using eclipse.